application/holder/src/env/fs/index.ts

```diff
@@ -74,27 +76,27 @@ export function getFolders(paths: string[]): string[] | Error {
     }
 }
 
-export function getFileEntities(files: string[]): File[] | Error {
+export async function getFileEntities(files: string[]): Promise<File[] | Error> {
     if (files.length === 0) {
         return [];
     } else {
         try {
-            return files
-                .map((filename: string) => {
-                    const entity: File | Error | undefined = getFileEntity(filename);
-                    if (entity instanceof Error) {
-                        throw entity;
-                    }
-                    return entity;
-                })
-                .filter((f) => f !== undefined) as File[];
+            const entities = [];
+            for(let i = 0; i < files.length; i++) {
+                const entity: File | Error | undefined = await getFileEntity(files[i])
+                if (entity instanceof Error) {
+                    throw entity
+                }
+                entities.push(entity);
+            }
+            return entities.filter((f) => f !== undefined) as File[];
         } catch (e) {
             return new Error(error(e));
         }
     }
 }
 
-export function getFileEntity(filename: string): File | undefined | Error {
+export async function getFileEntity(filename: string): Promise<File | undefined | Error> {
```

Comment:
as soon as it becomes `async` we cannot resolve promise with errors. 
`Promise<File | undefined | Error>` this is should be `Promise<File | undefined>` and error will go to rejection